packages/job-manager/src/batch/pool-metrics.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { injectable } from 'inversify';
+
+export interface PoolLoad {
+    activeTasks: number;
+    runningTasks: number;
+    pendingTasks: number;
+}
+
+export interface PoolMetricsInfo {
+    id: string;
+    maxTasksPerPool: number;
+    load: PoolLoad;
+}
+
+export interface PoolMetricsState {
+    lastPoolLoad?: PoolLoad;
+    lastTasksIncrementCount?: number;
+    processingSpeed?: number;
+}
+
+@injectable()
+export class PoolMetrics {
```

Comment:
how about TaskIncrementCountProvider?
and the method name could be:
getNextTaskIncrementCount. 
poolState field as preivate & is updated whenever getNextTaskIncrementCount is called.

Reply:
updated